_posts/2024-01-05-devcontainer-platform-agnostic-team.md

```diff
@@ -0,0 +1,68 @@
+
+Running all of the Pytests is also just a two-click endeavor: Click the `Testing` icon on the right-hand side, and the Play button to run all the specs.
+
+Although I was nervous about joining a team without other Mac engineers and using Visual Studio Code for the first time, being in this containerized developer environment and having both debugging and testing built in to the editor has been a time-saving and stress-relieving work environment. The Engineering team uses this Dev Container set up for all of our projects, including internal Compiler projects like this website itself as well, which, by the way, is [open source on GitHub](https://github.com/compilerla/compiler.la/).
+
+For a more basic step-by-step tutorial, refer to the official [VS Code Dev Containers tutorial](https://code.visualstudio.com/docs/devcontainers/tutorial) and the official [development container spec](https://containers.dev/overview). For more technical detail on Compiler’s specific implementation of Dev Containers for the Benefits application, refer to the application documentation on [Docker dynamic ports](https://docs.calitp.org/benefits/development/docker-dynamic-ports/) and [working with Dev Containers](https://docs.calitp.org/benefits/development/).
+
+
+## Benefits of a platform-agnostic team
+
+The benefits of having all engineers running and writing code from the same containerized environment and using the same code editor are quite clear: Less configuration headache when one engineer’s local environment has a seemingly un-replicable bug. But as I learned as I kept working on the Compiler team, I could see even more benefits of having this kind of environment supporting a team that has engineers using all major operating platforms.
+
+From the perspective of a team and company, especially one that works on open source government software, being able to support engineers who use Windows, Linux and Mac allows the company to recruit people from a wider variety of industries and experiences. I didn’t have to, for example, retrain and learn the Windows environment to work on government apps, which often are built by Windows-only teams.
+
+As an engineer who focuses primarily on the front-end user experience, I feel that our team is stronger because we are testing the app daily on Windows, Linux and Mac, across various browsers. I no longer have to click through an emulator took like BrowserStack to check how the app looks and feels on Windows. Windows users aren’t second-class, and vice versa. A common criticism of all-Apple engineering and design teams is that they are not working on the hardware that most Americans can afford and use, and that’s not a problem here at Compiler.
```

Comment:
```suggestion
As an engineer who focuses primarily on the front-end user experience, I feel that our team is stronger because we are testing the app daily on Windows, Linux and Mac, across various browsers. I no longer have to click through an emulator tool like BrowserStack to check how the app looks and feels on Windows. Windows users aren’t second-class, and vice versa. A common criticism of all-Apple engineering and design teams is that they are not working on the hardware that most Americans can afford and use, and that’s not a problem here at Compiler.
```